toolchain/parse/context.cpp

```diff
@@ -309,13 +279,16 @@ auto Context::DiagnoseOperatorFixity(OperatorFixity fixity) -> void {
     // Infix operators must satisfy the infix operator rules.
     if (!IsLexicallyValidInfixOperator()) {
       CARBON_DIAGNOSTIC(BinaryOperatorRequiresWhitespace, Error,
-                        "whitespace missing {0} binary operator", RelativeLoc);
-      emitter_->Emit(*position_, BinaryOperatorRequiresWhitespace,
-                     tokens().HasLeadingWhitespace(*position_)
-                         ? RelativeLoc::After
-                         : (tokens().HasTrailingWhitespace(*position_)
-                                ? RelativeLoc::Before
-                                : RelativeLoc::Around));
+                        "whitespace missing {0:=-1:before|=0:around|=1:after} "
```

Comment:
This slightly makes me want
```suggestion
                        "whitespace missing {0:<0:before|=0:around|>0:after} "
```
but it's probably not worth adding for this one use.

Reply:
FWIW I would still lean towards the specific values, since they're set below. I just felt -1/0/1 was easier to mentally map here than 0/1/2, but I could switch if desired